manager/src/main/java/com/redhat/service/bridge/manager/BridgesServiceImpl.java

```diff
@@ -181,23 +187,47 @@ public BridgeResponse toResponse(Bridge bridge) {
         return response;
     }
 
-    private void createTopicAndServiceAccount(String bridgeId) {
+    private void createTopicAndGrantAccessFor(String bridgeId) {
         if (!rhoasEnabled) {
             return;
         }
-        String topicName = String.format("ob-%s", bridgeId);
-        String serviceAccountName = String.format("ob-%s-consumer", bridgeId);
-        TopicAndServiceAccountRequest request = new TopicAndServiceAccountRequest(topicName, serviceAccountName);
         try {
-            rhoasClient.createTopicAndConsumerServiceAccount(request).await().atMost(Duration.ofSeconds(rhoasTimeout));
+            NewTopicInput newTopicInput = new NewTopicInput()
+                    .name(topicNameFor(bridgeId))
+                    .settings(new TopicSettings().numPartitions(1));
+
+            rhoasClient.createTopicAndGrantAccess(newTopicInput, rhoasOpsAccountClientId, RhoasTopicAccessType.CONSUMER_AND_PRODUCER)
+                    .await().atMost(Duration.ofSeconds(rhoasTimeout));
         } catch (CompletionException e) {
-            String msg = "Failed creating topic and service account for bridge " + bridgeId;
+            String msg = "Failed creating topic and granting access for bridge " + bridgeId;
             LOGGER.warn(msg, e);
             throw new InternalPlatformException(msg, e);
         } catch (TimeoutException e) {
-            String msg = "Timeout reached while creating topic and service account for bridge " + bridgeId;
+            String msg = "Timeout reached while creating topic and granting access for bridge " + bridgeId;
             LOGGER.warn(msg, e);
             throw new InternalPlatformException(msg, e);
         }
     }
+
+    private void deleteTopicAndRevokeAccessFor(String bridgeId) {
+        if (!rhoasEnabled) {
+            return;
+        }
+        try {
+            rhoasClient.deleteTopicAndRevokeAccess(topicNameFor(bridgeId), rhoasOpsAccountClientId, RhoasTopicAccessType.CONSUMER_AND_PRODUCER)
+                    .await().atMost(Duration.ofSeconds(rhoasTimeout));
+        } catch (CompletionException e) {
+            String msg = "Failed deleting topic and revoking access for bridge " + bridgeId;
+            LOGGER.warn(msg, e);
+            throw new InternalPlatformException(msg, e);
+        } catch (TimeoutException e) {
+            String msg = "Timeout deleting topic and revoking access for bridge " + bridgeId;
+            LOGGER.warn(msg, e);
```

Comment:
@kostola Let's just throw the `Exception` and not `LOGGER.warn` the same message. We'll end up with duplicate stack traces in the logs as `InternalPlatformExceptionMapper` additionally logs the error.

Reply:
Done :+1: